doc/setup-development-environment.md

```diff
@@ -0,0 +1,167 @@
+# HOWTO: Setup your Development Environment
+
+## Linux
+
+Install the dependencies needed for your distribution.  The best documentation
+are the Docker files used for the continuous integration builds.  For example,
+if you are planning to develop on Ubuntu open
+[Dockerfile.ubuntu](../ci/Dockerfile.ubuntu), there you will find the tools to
+install:
+
+```Dockerfile
+RUN apt-get update && apt-get install -y \
+  automake \
+  build-essential \
+  clang \
+  ... # More stuff ommitted.
+```
+
+Run the same commands as `root` in your workstation and your environment will
+be ready.
+
+### Installing Docker
+
+You may want to [install Docker](https://docs.docker.com/engine/installation/),
+this will allow you to use the build scripts to test on multiple platforms,
+as described in [CONTRIBUTING.md](../CONTRIBUTING.md).
+
+## Windows
+
+If you mainly use Windows as your development environment you need to install
+a number of tools.  We use [chocolatey](https://www.chocolatey.com) to drive the
+installation, so you would need to install it first.  This needs to be executed
+in a `cmd.exe` shell, running as the `Administrator`:
+
+```commandline
+c:> @"%SystemRoot%\System32\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -ExecutionPolicy Bypass -Command "iex (
+(New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))" && SET "PATH=%PATH%;%ALLUSERSPROFILE%\chocolatey\bin"
+```
+
+Then you can install the dependencies in the same shell:
+```commandline
+c:> choco install -y cmake git cmake.portable activeperl ninja golang yasm putty
+c:> choco install -y visualstudio2017community
+c:> choco install -y visualstudio2017-workload-nativedesktop
+c:> choco install -y microsoft-build-tools
+```
+
+The previous installation should create a
+`Developer Command Prompt for VS 2017` entry in your "Windows" menu, use that
+entry to create a new shell.
+In that shell install `vcpkg` the Microsoft supported ports for many Open Source
```

Comment:
[...] `vcpkg`, the Microsoft-supported [...]

Reply:
Fixed.